h2o-r/h2o-package/R/targetencoder.R

```diff
@@ -38,14 +38,14 @@
 h2o.targetencoder <- function(x,
                               y,
                               training_frame,
+                              model_id = NULL,
```

Comment:
@sebhrusen I had to generate files with accordance to backend changes ( git diff check would fail otherwise). I think that this should be considered as a API breaking changes right? it looks like in our tests we always use named parameters
```
target_encoder <- h2o.targetencoder(training_frame = data, x = encoded_columns, y = "survived")
```
instead of relying just on order of parameters. But it means someone who does rely on order of specifying... will be in trouble. Or is there a convention in R that you should use named parameters style?

Reply:
technically, it is a potential breaking change.
practically, I think that R users (like Python users) use positional params only for functions with very few params, or for the first 2-3 params max: common usage is to use named params.
@ledell to confirm?

Reply:
In my opinion, it's ok to add this in the middle to be consistent with other modeling functions.  It's not generally advised to do this, because it will break code of people who relied only on the order of the arguments, but we probably have very few people using this function yet, considering our User Guide doesn't even have the up-to-date examples for TE.

Reply:
@ledell @sebhrusen ok then. Going to merge after tests are passed.